php/queryads.php

```diff
@@ -0,0 +1,34 @@
+<?php
+ob_end_flush();
+ini_set("output_buffering", "0");
+ob_implicit_flush(true);
+header('Content-Type: text/event-stream');
+header('Cache-Control: no-cache');
+
+function echoEvent($datatext) {
+    echo "data: ".implode("\ndata: ", explode("\n", $datatext))."\n\n";
+}
+
+// Test if domain is set
+if(isset($_GET["domain"]))
+{
+    // Remove illegal characters
+    $url = filter_var($_GET["domain"], FILTER_SANITIZE_URL);
+    // Is this a valid domain?
+    if(!filter_var(gethostbyname($url), FILTER_VALIDATE_IP))
+    {
+        echoEvent("Invalid domain!");
+        die();
+    }
+}
+else
+{
+    echoEvent("No domain provided");
+    die();
+}
+
+$proc = popen("sudo pihole -q ".$url, 'r');
```

Comment:
I've tried different combinations of stuff to get past the filtering and cause a security leak, but have been unsuccessful (that's a good thing). I'd like some other people who might know more tricks to try though, so we make sure that we're not opening any security holes in our users' systems.